libs/surfaces/websockets/server.cc

```diff
@@ -59,9 +60,29 @@ WebsocketsServer::WebsocketsServer (ArdourSurface::ArdourWebsockets& surface)
 	_lws_proto[0] = proto;
 	memset (&_lws_proto[1], 0, sizeof (lws_protocols));
 
+	// '/' is served by a static index.html file in the surface data directory
+	// inside it there is a 'builtin' subdirectory that contains all built-in
+	// surfaces so there is no need to create a dedicated mount point for them
+	// list of surfaces is available as a dynamically generated json file
+	memset (&_lws_mnt_index, 0, sizeof (lws_http_mount));
+	_lws_mnt_index.mountpoint      = "/";
+	_lws_mnt_index.mountpoint_len  = strlen (_lws_mnt_index.mountpoint);
+	_lws_mnt_index.origin_protocol = LWSMPRO_FILE;
+	_lws_mnt_index.origin          = _resources.index_dir ().c_str ();
+
+	// user defined surfaces in the user config directory
+	memset (&_lws_mnt_user, 0, sizeof (lws_http_mount));
+	_lws_mnt_user.mountpoint      = "/user";
```

Comment:
Isn't there a trailing slash missing?  `/user` (file) vs  `/user/` (dir)

Instead of `/user/` perhaps use `/~/` ? To avoid hardcoding English?

Reply:
You are right, directories should be represented using a trailing slash.

The `/~/` suggestion is cleaner but note there is also a directory for serving built-in surfaces (residing within Ardour's data directory) hardcoded as `/builtin/`. This is not visible in code but in the filesystem (see `{GIT_ROOT}/share/web_surfaces/builtin`). Originally the code implemented a third mount point for these but later I decided that was unneeded complexity because the `/` mount point can already serve `/builtin/` by relying on the filesystem structure. Web clients will make requests to either `/~/...` or `/builtin/...` depending the surface location on disk. So there is a further English hardcoded string. What do you think?

Reply:
When reading over it, I thought  `/` is used for bundled/built-in content, and `~/` for per-user surfaces.

Anyway this is not a blocker, for this PR. Keeping `/user/` is fine with me.
If needed, we can still change this later (pre-release) after further discussion.

Reply:
Seems libwebsockets does not like the trailing slash; appending it breaks the links to user defined data (404). Cannot find docs for the expected format or examples where the mount point is not the trivial "/". Guess it is not a blocker as well.

https://libwebsockets.org/lws-api-doc-master/html/structlws__http__mount.html#aa2391bfcada0b7a290b3c6651f64586c 

https://github.com/warmcat/libwebsockets/blob/master/minimal-examples/http-server/minimal-http-server/minimal-http-server.c